packages/react/menu/src/Menu.tsx

```diff
@@ -361,6 +471,89 @@ const MenuContentImpl = React.forwardRef((props, forwardedRef) => {
 
 MenuContent.displayName = CONTENT_NAME;
 
+/* -------------------------------------------------------------------------------------------------
+ * MenuSubContent
+ * -----------------------------------------------------------------------------------------------*/
+
+const SUB_CONTENT_NAME = 'MenuSubContent';
+
+type MenuSubContentOwnProps = Polymorphic.Merge<
+  Omit<
+    Polymorphic.OwnProps<typeof MenuContentImpl>,
+    'portalled' | 'disableOutsidePointerEvents' | 'disableOutsideScroll' | 'side' | 'trapFocus'
+  >,
+  {
+    /**
+     * Used to force mounting when more control is needed. Useful when
+     * controlling animation with React animation libraries.
+     */
+    forceMount?: true;
+  }
+>;
+type MenuSubContentPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof MenuContentImpl>,
+  MenuSubContentOwnProps
+>;
+
+const MenuSubContent = React.forwardRef((props, forwardedRef) => {
+  const { forceMount, dir, ...contentProps } = props;
+  const context = useMenuContext(SUB_CONTENT_NAME);
+  const subMenuContext = useMenuSubContext(SUB_CONTENT_NAME);
+  const direction = dir || context.dir;
+  const timerRef = React.useRef(0);
+
+  React.useEffect(() => {
+    return () => window.clearTimeout(timerRef.current);
+  }, []);
```

Comment:
When we have a part that renders the `Presence` component, we move all the logic that isn't associated with visibility into a `Impl` part so that none of the logic is running in the background unnecessarily when `present` is false.

This part tripped me up because I thought you were expecting it to clear the timeout when it closes, which wouldn't be the case here as this part would always be mounted when it's closed. But reading more, I don't _think_ it matters in this particular case but it would be good to follow our convention just to prevent that mental mapping in future.

Usually our `<Presence />` parts look pretty bare:

![image](https://user-images.githubusercontent.com/175330/118334783-2c253800-b506-11eb-9f83-aea7515bb6ea.png)

Reply:
Thanks for this btw, it's a good habit and useful to remember!